File: etl-common/src/test/scala/org/bigdate/etl/common/scala/test/spark/executors/middle/DirtyMiddleExecutor.scala
package org.bigdate.etl.common.scala.test.spark.executors.middle

import org.apache.spark.rdd.RDD
import org.apache.spark.sql.{DataFrame, SparkSession}
import org.bigdata.etl.common.annotations.ETLExecutor
import org.bigdata.etl.common.executors.MiddleExecutor
import org.bigdate.etl.common.scala.test.spark.configs.DirtyConfig
import org.slf4j.{Logger, LoggerFactory}

/**
 * Author: GL
 * Date: 2022-05-05
 */
@ETLExecutor("dirty")
class DirtyMiddleExecutor extends MiddleExecutor[SparkSession, RDD[String], DataFrame, DirtyConfig] {

  val log: Logger = LoggerFactory.getLogger(classOf[DirtyMiddleExecutor])

  override def init(engine: SparkSession, config: DirtyConfig): Unit = {
    log.info("DirtyMiddle init, config: {}", config)
  }

  override def process(engine: SparkSession, value: RDD[String], config: DirtyConfig): DataFrame = {
    import engine.implicits._
    value.toDF()
  }

  override def close(engine: SparkSession, config: DirtyConfig): Unit = {
    log.info("DirtyMiddleExecutor close, config: {}", config)
  }

  override def check(config: DirtyConfig): Boolean = {
    log.info("DirtyMiddleExecutor check, config: {}", config)
    true
  }
}

File: etl-common/src/test/java/org/bigdata/etl/common/executors/middle/DirtyMiddleExecutor.java
package org.bigdata.etl.common.executors.middle;

import lombok.extern.slf4j.Slf4j;
import org.apache.spark.SparkContext;
import org.apache.spark.rdd.RDD;
import org.bigdata.etl.common.annotations.ETLExecutor;
import org.bigdata.etl.common.configs.DirtyConfig;
import org.bigdata.etl.common.executors.MiddleExecutor;

import java.util.Collection;
import java.util.Optional;


/**
 * Author: GL
 * Date: 2022-04-22
 */
@Slf4j
@ETLExecutor("dirty")
public class DirtyMiddleExecutor implements MiddleExecutor<SparkContext, RDD<String>, RDD<String>, DirtyConfig> {

    @Override
    public void init(SparkContext engine, DirtyConfig config) {
        log.info("DirtyMiddle init, config: {}", config);
    }

    @Override
    public RDD<String> process(SparkContext engine, RDD<String> value, DirtyConfig config) {
        log.info("DirtyMiddle process, config: {}, middle collect: {}", config, value.collect());
        return value;
    }

    @Override
    public void close(SparkContext engine, DirtyConfig config) {
        log.info("DirtyMiddleExecutor close, config: {}", config);
    }

    @Override
    public boolean check(DirtyConfig config) {
        log.info("DirtyMiddleExecutor check, config: {}", config);
        return true;
    }
}
